client/state/slices/teamSlice.ts

```diff
@@ -0,0 +1,24 @@
+import { PayloadAction, createSlice } from "@reduxjs/toolkit";
+import { RootState } from "../store";
+import { defaultTeam, Team } from "@/types/teamTypes";
+
+const teamSlice = createSlice({
+  name: "team",
+  initialState: defaultTeam,
+  reducers: {
+    setTeam: (_, action: PayloadAction<Team>) => action.payload,
+    clearTeam: () => defaultTeam,
+  },
+});
+
+export const selectTeamId = (state: RootState) => state.team.id;
+
+export const selectIsTeamRegistered = (state: RootState): boolean =>
+  state.team.id !== undefined &&
+  state.team.abbreviation !== undefined &&
+  state.team.logoUrl !== undefined &&
+  state.team.name !== undefined;
```

Comment:
We could use that `isNotNil` util func here

Do we need to check if all things are undefined or can we just check for one of these (team.id) optimistically? I assume the others should not exist if we don't have an id